rust/src/connection.rs

```diff
@@ -7,7 +7,11 @@ use anyhow::Result;
 pub async fn keep_alive() -> Result<()> {
     let mut connected = false;
     loop {
-        let is_connection_lost_maybe = !is_first_channel_usable()?;
+        let channel_details = wallet::get_first_channel_details();
+
+        let is_connection_lost_maybe = channel_details.is_some()
+            && !is_first_channel_usable()
+            && channel_details.unwrap().short_channel_id.is_some();
```

Comment:
Let me try to understand this:

`is_connection_lost_maybe` describes that the connection is (maybe) *lost*, and it is lost if we have a channel (`channel_details.is_some()`) and this channel is not usable (`!is_first_channel_usable()`) and we can retrieve the short channel id from said channel (`channel_details.unwrap().short_channel_id.is_some()`)...?

I think the last condition does not make sense - I think this should be:

```
 let is_connection_lost_maybe = channel_details.is_some()
            && (!is_first_channel_usable()
            || !channel_details.unwrap().short_channel_id.is_some());
```